package.json

```
@@ -6,7 +6,7 @@
     "docs:dev": "yarn workspace docs dev",
     "docs:start": "yarn workspace docs start",
     "docs:create-playground": "yarn workspace docs create-playground",
-    "docs:api": "yarn docs:api:build",
+    "docs:api": "NODE_OPTIONS=--max_old_space_size=4096 yarn docs:api:build",
```

```suggestion
    "docs:api": "NODE_OPTIONS=--max-old-space-size=4096 yarn docs:api:build",
```

As per https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes

---

Is this change relevant locally as well? 🤔 
If it does speed up local runs, then sure, let's go with it. 👍 
Otherwise, have you considered [this approach:](https://github.com/mui/mui-x/blob/master/.circleci/config.yml#L227)?

---

This option isn't about spead, it's about the heap size avaialbe for JS objects. In that sense, I'm surprised that this problem doesn't also affect local runs. If we're OOM in the CI, we should be OOM locally. My guess would be the GC is doing at least part of its work multi-threadeded and can run more efficiently on local dev laptops with more cores/power, that's why the heap limit is not reached. But yes I would put the option locally as well, it's not doing any harm.

---

Could it be that CircleCI Node defaults to 2GB of heap space, whereas locally it has a 4GB heap space limit? 🤔 
If I remember correctly when I was debugging OOM issues for typescript, the CI had a 2GB heap limit, but I can't 100% confirm it, because the pipeline information seems to have been removed already.
And we still were on node 14 on CI back then. 🤔 
https://github.com/mui/mui-x/pull/8796/commits/b84f2fb34c3cb4db72ce597c70ebed69467249ec
https://github.com/mui/mui-x/pull/8796

---

> I'm surprised that this problem doesn't also affect local runs. If we're OOM in the CI, we should be OOM locally.

Could it be that the default value for `--max-old-space-size` depends on the memory size?